### PR TITLE
Minor build system fixes.

### PR DESCRIPTION
**Pull request by CarloWood**
This wasn't entirely correctly configured...

### CODE REVIEW

**Comment by gennadiycivil** [comment]
@coryan any comments on this ? Thank you!

**Comment by coryan** [comment]
LGTM, and the build with autotools seems happy:

https://travis-ci.org/google/googletest/jobs/500203907

**Comment by CarloWood** [comment]
Hmm, now I'm confused... In that link it says:
aclocal: warning: couldn't open directory 'm4': No such file or directory
...
libtoolize: Consider adding `AC_CONFIG_MACRO_DIR([m4])' to configure.ac

while to whole point of my patch was to get rid of any of such warnings.
(I don't get them anymore when running autoreconf).

It seems that AC_CONFIG_MACRO_DIRS (with S) was added to automake 1.13
and autoconf 2.70 (while AC_CONFIG_MACRO_DIR is only in autoconf as of 2.58).

What are the versions used on travis?

**Comment by CarloWood** [comment]
This is what I get locally (with my patch):

`
sean:~/projects/aicxx/ai-evio-testsuite/ai-evio-testsuite/googletest>autoreconf -v
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: adding subdirectory googletest to autoreconf
autoreconf: Entering directory `googletest'
autoreconf: running: aclocal -I m4
autoreconf: configure.ac: not running libtoolize: --install not given
autoreconf: running: /usr/bin/autoconf
autoreconf: running: /usr/bin/autoheader
autoreconf: running: automake --no-force
autoreconf: Leaving directory `googletest'
autoreconf: configure.ac: adding subdirectory googlemock to autoreconf
autoreconf: Entering directory `googlemock'
autoreconf: Leaving directory `googlemock'
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Autoheader
autoreconf: Leaving directory `.'
`

**Comment by gennadiycivil** [comment]
@CarloWood is this ready to merge? ( Not sure based on the comments)

**Comment by coryan** [comment]
Hold on. I may have read that log incorrectly.

**Comment by coryan** [comment]
@CarloWood As I recall, Travis is based on Ubuntu 14.04 (or 16.04 if you enable `dist: trusty`):

https://docs.travis-ci.com/user/reference/precise/

Ubuntu 14.04 ships with autoconf-2.69 and automake-1.14. Which may explain the problem?

I think @gennadiycivil needs to decide what distributions he wants to support.  Ubuntu 14.04 is going EOL ~soon (April I believe?), but there are other distros, such as CentOS 7 that remain "popular" (Okay, widely used), and ship with the same versions of autoconf and automake.

Just my $0.02.

**Comment by CarloWood** [comment]
Well, it works in either case (as you see). The provided patch is just a little bit more modern and removes warnings when you use tools that aren't about to be no longer supported next month.

**Comment by CarloWood** [comment]
In my opinion this can be merged. The two remaining warnings (although it surprised me at first to see ANY warnings at all) are completely harmless.

File: .gitignore [hunk 1]
```diff
@@ -35,6 +35,7 @@ googletest/m4/ltoptions.m4
 googletest/m4/ltsugar.m4
 googletest/m4/ltversion.m4
 googletest/m4/lt~obsolete.m4
+googlemock/m4
```
Comment (ngie-eign) [inline .gitignore]: Some things are missing from .gitignore. Here are a full list of files (in sorted order) that should be added:

```
$ git diff .gitignore 
diff --git a/.gitignore b/.gitignore
index 6209bf66..5afc957c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -24,17 +24,54 @@ Win32-Release/
 x64-Debug/
 x64-Release/
 
+# Ignore binaries built by tests in tree.
+googlemock/gmock-actions_test
+googlemock/gmock-cardinalities_test
+googlemock/gmock-function-mocker_test
+googlemock/gmock-generated-actions_test
+googlemock/gmock-generated-function-mockers_test
+googlemock/gmock-generated-matchers_test
+googlemock/gmock-internal-utils_test
+googlemock/gmock-matchers_test
+googlemock/gmock-more-actions_no_exception_test
+googlemock/gmock-more-actions_test
+googlemock/gmock-nice-strict_test
+googlemock/gmock-port_test
+googlemock/gmock-spec-builders_test
+googlemock/gmock_ex_test
+googlemock/gmock_leak_test_
+googlemock/gmock_link_test
+googlemock/gmock_no_rtti_test
+googlemock/gmock_output_test_
+googlemock/gmock_stress_test
+googlemock/gmock_test
+googlemock/shared_gmock_test_
+
 # Ignore autoconf / automake files
+*.lo
+*.o
+.deps
+.dirstamp
+.libs
 Makefile.in
 aclocal.m4
-configure
-build-aux/
 autom4te.cache/
+build-aux/
+configure
+config.log
+config.status
+googlemock/lib/
+googlemock/libtool
+googlemock/m4/
+googlemock/scripts/gmock-config
+googletest/libtool
+googletest/lib/
 googletest/m4/libtool.m4
 googletest/m4/ltoptions.m4
 googletest/m4/ltsugar.m4
 googletest/m4/ltversion.m4
 googletest/m4/lt~obsolete.m4
+googletest/scripts/gtest-config
 
 # Ignore generated directories.
 googlemock/fused-src/
```

**Comment by ngie-eign** [comment]
> @CarloWood As I recall, Travis is based on Ubuntu 14.04 (or 16.04 if you enable `dist: trusty`):
> 
> https://docs.travis-ci.com/user/reference/precise/
> 
> Ubuntu 14.04 ships with autoconf-2.69 and automake-1.14. Which may explain the problem?
> 
> I think @gennadiycivil needs to decide what distributions he wants to support. Ubuntu 14.04 is going EOL ~soon (April I believe?), but there are other distros, such as CentOS 7 that remain "popular" (Okay, widely used), and ship with the same versions of autoconf and automake.
> 
> Just my $0.02.

@CarloWood's fixes in the branch unbreak compiling googletest with autoconf on FreeBSD, so I suspect that this issue is much more widespread than Ubuntu Trusty.

The autoconf project has started deprecating some singularly named APIs in favor of plurally named ones, e.g., `AC_CONFIG_MACRO_DIR` -> `AC_CONFIG_MACRO_DIRS`. I think that moving forward with the change is in Google's best interests if they wish to maintain the autotools port.

As a sidenote, I need this commit in order to help binaries built by autotools, as the next step would be introducing an upstream provided `ax_pthread.m4` to fix the aging `acx_pthread.m4`, which doesn't properly compile with modern compilers, e.g., clang++.

**Comment by gennadiycivil** [comment]
> @CarloWood As I recall, Travis is based on Ubuntu 14.04 (or 16.04 if you enable `dist: trusty`):
> 
> https://docs.travis-ci.com/user/reference/precise/
> 
> Ubuntu 14.04 ships with autoconf-2.69 and automake-1.14. Which may explain the problem?
> 
> I think @gennadiycivil needs to decide what distributions he wants to support. Ubuntu 14.04 is going EOL ~soon (April I believe?), but there are other distros, such as CentOS 7 that remain "popular" (Okay, widely used), and ship with the same versions of autoconf and automake.
> 
> Just my $0.02.

coryan@ thank you! The question is "what distros and what build systems" I believe. We do have bazel as the "primary" build system and CMake and automake as "community supported" . I would prefer *soon* to drop one of them and just have one "primary" and one "secondary, community-supported" one ( CMake I think it much more popular ).

**Pull request merged**